roofit/multiprocess/src/Messenger.cxx

```diff
@@ -479,8 +479,6 @@ void Messenger::set_send_flag(zmq::send_flags flag)
 
 // -- MASTER - WORKER COMMUNICATION --
 
-void Messenger::publish_from_master_to_workers() {}
```

Comment:
While you are changing this file, can you please also fix this warning that I get when building ROOT?

```
/home/rembserj/root-builds/master-deb-info/src/roofit/multiprocess/src/Messenger.cxx: In constructor ‘RooFit::MultiProcess::Messenger::Messenger(const RooFit::MultiProcess::ProcessManager&)’:
/home/rembserj/root-builds/master-deb-info/src/roofit/multiprocess/src/Messenger.cxx:79:15: warning: variable ‘rc’ set but not used [-Wunused-but-set-variable]
   79 |          auto rc = zmq_setsockopt(*mq_push_, ZMQ_SNDHWM, &hwm, sizeof hwm);
      |               ^~
/home/rembserj/root-builds/master-deb-info/src/roofit/multiprocess/src/Messenger.cxx:163:15: warning: variable ‘rc’ set but not used [-Wunused-but-set-variable]
  163 |          auto rc = zmq_setsockopt(*mq_push_, ZMQ_SNDHWM, &hwm, sizeof hwm);
      |               ^~
/home/rembserj/root-builds/master-deb-info/src/roofit/multiprocess/src/Messenger.cxx:202:15: warning: variable ‘rc’ set but not used [-Wunused-but-set-variable]
  202 |          auto rc = zmq_setsockopt(*mw_sub_, ZMQ_RCVHWM, &hwm, sizeof hwm);
  ```

That's just one observation I made when building, more detailed review will follow.

Reply:
Ah yes, this is because of the asserts which only generate code in debug builds. Weird that this didn't trigger CI warnings before! Will fixup the commit that touches `Messenger` to avoid these warnings.

Reply:
Changed to type-safe C++ versions that simply throw on rc != 0 internally.